File: leetcode/odd-even-linked-list.py
# Definition for singly-linked list.
class ListNode:
    def __init__(self, val=0, next=None):
        self.val = val
        self.next = next

class Solution:
    def oddEvenList(self, head: ListNode) -> ListNode:
        if head == None or head.next == None:
          return head

        odd = head
        even = head.next
        evenHead = even

        while even != None and even.next != None:
          odd.next = even.next
          odd = odd.next
          even.next = odd.next
          even = even.next

        odd.next = evenHead

        return head

def printLinkedList(head: ListNode, prev=''):
  curr = head
  if len(prev) > 0:
    print(prev)
  while (curr != None):
    print(curr.val)
    curr = curr.next


if __name__ == '__main__':
  head = ListNode(val = 1, next=ListNode(2, next=ListNode(3)))

  printLinkedList(head)

  # print('head', head.val)

  solution = Solution()

  # print

  ans = solution.oddEvenList(head)


  printLinkedList(ans, prev='solution')

# a -> (b) -> c -> (d) -> NULL
# odd: a -> c
# even: b

File: leetcode/odd-even-linked-list.cpp
#include <bits/stdc++.h>

using namespace std;

struct ListNode {
  int val;
  ListNode *next;
  ListNode() : val(0), next(nullptr) {}
  ListNode(int x) : val(x), next(nullptr) {}
  ListNode(int x, ListNode *next) : val(x), next(next) {}
};

ListNode* oddEvenList(ListNode* head) {

  if (head == nullptr || head->next ==nullptr) {
    return head;
  }

  ListNode* ans = head, curr = head;
  ListNode* odd;


  while (curr.next != nullptr && curr.next.next != nullptr)

}

int main() {




  return 0;
}


/**

1 -> 2 -> 3 -> 4 -> 5
o         

**/